### DISCUSSION TITLE
ESP32 with MPR121 Touch switch sensor with Micropython

### DISCUSSION THREAD

**Discussion by kdavies-kd** [micropython/micropython#15178] [ESP32] [answered]
Hi, this is my first IOT project and I feel I must extend a warm thank you, as although I found it challenging, I have been able to get the first bread board design out from my head to a working proof of concept with MicroPython.
I simply touch the sensor channel 1 and LED 1 comes on.
But now to add some usability I need some help and guidance.
How would I create a simple desktop app to show the status of the LED's from GPIO readings? and also set the GPIO.
A simple canvas with a sliding switch on or off, working trough the connected USB port.
The target pc will generally have no access to the internet.
And I also need help to figure out the settings to use the MPR121 as a capacitance touch switch. At the moment I am just touching the contacts. I want to be able to operate them from underneath some thin Plasticard.
I have been able to get the irq-key-down.py script working from MPR121.py driver.
Thanks in advance.

**Comment by peterhinch** [comment, marked as answer]
The ESP32 has a built in [capacitive touch sensor](https://docs.micropython.org/en/latest/esp32/quickref.html#capacitive-touch). It is capable of working through a thin nonconductive film.

**Reply by kdavies-kd** [reply]
Thanks for that, missed it completely, remapping data pins accordingly.
I am working on the Arduino Nano ESP32-S3
Thank you for your prompt informative reply

**Comment by kdavies-kd** [comment]
Hi
I have managed to get a working model for my 6 channel switching device 
1. with the MPR121 touch switch sensors working as expected not able to use the built in ones. with
2. 6 LED channel indicators. 
And now I need some more help and guidance.
How would I create a simple desktop app to communicate with the esp32-s3?
Could this work by using the existing USB port?
1. to show the status of the LED's from GPIO readings? and
3. also set the GPIO pins.
A simple canvas with a sliding switch on or off, working trough the connected USB port.
The target pc will generally have no access to the internet.
I have found dearpygui, considering this for the makings of a desktop app.
has usb on-the-go been implemented in micro python on the esp32-s3?
